engines/dradis-api/lib/dradis/ce/api/api_token_strategy.rb

```diff
@@ -0,0 +1,50 @@
+module Dradis::CE::API
+  class ApiTokenStrategy < ::Warden::Strategies::Base
+    attr_reader :email, :password
+
+    # This strategy should be applied when we are requesting /api/
+    def valid?
+      !!(request.path_info =~ /\A\/api\//)
```

Comment:
The goal of `valid?` is to guard early against unmet conditions for the strategy.

I think that L11, the presence of `token` is a prerequisite for the strategy to apply and should be moved to the `valid?` method.

That way it becomes explicit that a token is required, and the the `authenticate!` method becomes cleaner.